v2/tools/generator/internal/astmodel/resource_type.go

```diff
@@ -621,23 +623,47 @@ func (resource *ResourceType) AsDeclarations(codeGenerationContext *CodeGenerati
 	}
 
 	var declarations []dst.Decl
+	var errs []error
 	declarations = append(declarations, resourceDeclaration)
 	declarations = append(declarations, resource.InterfaceImplementer.AsDeclarations(codeGenerationContext, declContext.Name, nil)...)
-	declarations = append(declarations, resource.generateMethodDecls(codeGenerationContext, declContext.Name)...)
+
+	decls, err := resource.generateMethodDecls(codeGenerationContext, declContext.Name)
+	if err != nil {
+		errs = append(errs, err)
```

Comment:
Can we panic here instead? I don't see any loop around where we need to aggregate errors

Reply:
Fair call, will change. 

When I wrote the code, I expected the other lines contributing to the `declarations` slice would also end up returning errors, but they didn't.